src/utils/collection/apply.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const arraySize = require('../array').size
+const isMatrix = require('../collection/isMatrix')
+const IndexError = require('../../error/IndexError')
+
+/**
+ * Apply function that takes an array and returns a Scalar
+ * along a given axis of a matrix or array.
+ * Returns a new matrix or array with one less dimension.
+ * @param {Array | Matrix} mat
+ * @param {number} dim
+ * @param {Function} callback
+ * @return {Array | Matrix} res
+ */
+module.exports = function (mat, dim, callback) {
```

Comment:
Would this be better implemented as a typed function?

Reply:
Potentially. I was following the form of the other utility functions like reduce.js or deepMap.js which are not implemented as typed functions. I would be happy to make this change if you think it would be improve the quality of the function.